### PR TITLE
Milestone 3: Game logic

### PR DESCRIPTION
**Pull request by BrittanyBlake**
Our previous pull request for milestone 3 (pull request #3 ) was accidentally merged when we resolved the conflicts and merged the pull request for milestone 4 (pull request #4 ). We attempted to undo the merge and found that we were unable to do so, so we have linked the old pull request here: https://github.com/BrittanyBlake/tic-tac-toe/pull/3

As required by the TSE, we have implemented the following changes:
- Removed the put statement from the board.rb file and moved it into the main.rb file.
- Removed the error that was being raised in the board.rb file so that it no longer breaks the code.
- Changed attr_accessors to attr_readers and attr_writers

We look forward to your comments

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -54,39 +50,48 @@
 puts ' 1 0 | 1 1 | 1 2 '
 puts '-----+-----+----'
 puts ' 2 0 | 2 1 | 2 2 '
-
 new_game = Logic.new(player1, player2, symbol1, symbol2)
-
 game_over = false
 turn = 1
-
 until game_over == true
```
Comment (suretrust) [inline bin/main.rb]: This is the same as saying 
```ruby
until game_over
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job implementing some of the suggested changes. 🙌 Please see my comments on your code. 👇 
- [ ] The application shouldn't exit due to invalid input - it should rather inform the player about it and ask for their move again. From the previous review, this issue still persists, let's fix this.

> The application does exit due to invalid input without space (example: `qr`) - instead, it should inform the player about it and asks for their move again. Also, it should ask for correct input if I input something like this: `2 r` or `q r`.

- [ ] `[Optional]` Having to add a space between the input `0 0` is somehow a bad user experience, you could accept `00` without space instead.

**Kindly submit for another code review request once these issues have been resolved.** _Happy coding!_ 🎉

**Comment by ddddavis** [comment]
- Modified application, so it no longer exists during invalid input
- Improved user experience, so you no longer need to add space between user input
- Modified game_over condition

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
The project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team, 
### You did a great job so far:+1:
Still, there are some more changes required:
* [ ] The application should save the player's move and display the status of the board after that. Right now, it's hard to play your game if you can't see the field with free cells.
* [ ] Your game interface still asks the user to enter two space-separated numbers representing a position in the format `row col`. It needs to be changed since you removed space between user input.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Comment by BrittanyBlake** [comment]
As requested by TSE, we implemented the following changes:
-  Edited the input statement asking them to enter two space-separated numbers (however we decided to leave the phrase specifying the format is `row col` to ensure the user understands how the coordinates work).
- Added the missing puts keyword to ensure the updated board is displayed after each move.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

You have both done quite well on this.👍

However @davisdambis you have fewer commits than @BrittanyBlake  kindly note that Microverse supports an equal number of commits between student partners. Please correct this in subsequent milestones.

Please consider the charges, stated below.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

PLEASE TAKE NOTE OF THE BELOW STATED CHANGES👇

File: lib/board.rb [hunk 2]
```diff
@@ -3,43 +3,49 @@ class Board
...
+
   def [](position)
     row, col = position
     @grid[row][col]
   end
+
   def []=(position, value)
     row, col = position
     @grid[row][col] = value
   end
```
Comment (mikenath223) [inline lib/board.rb]: Kindly change this to use attr_reader/attr_writer, as this defeats the purpose of using attrs, refer to [this stack overflow post](https://stackoverflow.com/a/5046915/12217655) for tips, also [this article](https://www.rubyguides.com/2018/11/attr_accessor/) discusses how attr_reader/attr_writer is a shorter syntax.

**Comment by mikenath223** [comment]
In addition, consider changing, the attr_accessor used here to the appropriate attr_reader and/or attr_writer, depending on the use case.[Screenshot]👇
ice_screenshot_20200512-035020


### Your codebase looks clean and modular, thumbs up on that👍👍

**Comment by BrittanyBlake** [comment]
As requested by TSE, we implemented the following changes:

- We changed "attr_accessor :grid" to "attr_reader :grid attr_writer :grid" as requested by TSE.

     * However, we are a bit puzzled by this feedback. We used attr_reader and attr_writer in other parts of the project, so our not using it for grid was deliberate. According to the Microverse project guidelines, we are permitted to use att_accessor where it is necessary and we felt that it was necessary here. In our opinion, which was later validated by the article given by the TSE, attr_accessor should be used when we want to create both a getter and a setter, and since we did not have any other variables for which we needed to create the reader, we thought that this would be an appropriate situation in which to use attr_accessor. If this is not the case, please let us know in the subsequent review what the appropriate situation would be so that we do not repeat this error in the future.

File: lib/board.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 class Board
-  attr_accessor :grid
+  attr_reader :grid
+  attr_writer :grid
```
Comment (Dandush03) [inline lib/board.rb:3]: This line can be deleted since it has no use.

File: lib/logic.rb [hunk 4]
```diff
@@ -2,8 +2,8 @@
 require_relative '../lib/player.rb'
 
 class Logic
-  attr_reader :player1, :player2, :current_player
-  attr_accessor :board
+  attr_reader :player1, :player2, :current_player, :board
+  attr_writer :board
```
Comment (Dandush03) [inline lib/logic.rb:6]: This line has no use.

--- 
---

### Hope it helped to clear the issue

## Happy Coding :computer: :nerd_face: 

#### Daniel Laloush

**Comment by mikenath223** [comment]
> As requested by TSE, we implemented the following changes:
> 
> * We changed "attr_accessor :grid" to "attr_reader :grid attr_writer :grid" as requested by TSE.
>   
>   * However, we are a bit puzzled by this feedback. We used attr_reader and attr_writer in other parts of the project, so our not using it for grid was deliberate. According to the Microverse project guidelines, we are permitted to use att_accessor where it is necessary and we felt that it was necessary here. In our opinion, which was later validated by the article given by the TSE, attr_accessor should be used when we want to create both a getter and a setter, and since we did not have any other variables for which we needed to create the reader, we thought that this would be an appropriate situation in which to use attr_accessor. If this is not the case, please let us know in the subsequent review what the appropriate situation would be so that we do not repeat this error in the future.

This is noted, however kindly fix the other reference issue. Thanks.

File: lib/board.rb (hunk 3 above)
Comment (Dandush03) [reply lib/board.rb:3]: -------

-------

### Hope I helped this time

# Have A Nice Coding Day :computer: :nerd_face: 

#### Daniel Laloush

**Review by Dandush03 — COMMENTED** [review]
# Status: :red_circle: Required Changes :red_circle: 

## Hey! :wave: 

### You did a good job with your project :muscle: 

I see that you have some miss leading information regarding when or what is an attr writer. so I will do my best to briefly explain this and give you some examples. If you have any questions after this, try contacting me through slack.

Essentially when do we need an attr_reader?
Only when we directly manipulate our variable outside of the class. let us say if I have a class name board with a board variable inside of it (to make it suitable to this example) and another one name game. When I try to manipulate the variable "board" the following way
```
required 'board'

class Game
  def initialize
    board = new Board()
    board.board = 0
  end
end
```
then only under this example, I would need to set my `Board.board` to be an attr_writter but if I use it like this:
```
required 'board'

class Game
  def initialize
    board = new Board()
    board.changeBoard = 0
  end
end
```
and the `board.changeBoard` is a function that changes the state of the variable inside the class. then there is no need to use the attr_writer on this variable. 

I hope it helps to clear the issue. 

### Now let us dig in into your project.

#### :red_square: This is very important and I should mark this project invalid because of this, but since you got so far I will let you go with a warning and asking you to change it. :red_square: 
 > Please change back your files to it default value, there is no way the `Layout/LineLength:` will let you have 1200 character in one line

#### Following my explanation and after strongly review the code I will mark bellow what 2 attr_writer that aren't needed, in any case, test your code after making a new submit
##### Note: If by any mean you need to use attr_reader and attr_writer then you most use attr_accessor

**Pull request merged**